core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/ExpressionValidator.scala

```diff
@@ -0,0 +1,48 @@
+package org.broadinstitute.dsde.rawls.expressions
+
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.jobexec.MethodConfigResolver
+import org.broadinstitute.dsde.rawls.model.{AttributeString, ErrorReport, MethodConfiguration, ValidatedMethodConfiguration}
+import spray.http.StatusCodes
+
+import scala.util.Try
+
+object ExpressionValidator {
+  // inputsToParse is parameterized in order to validate in the presence or absence of the associated Method
+  // presence: inputs which are both empty and optional are pre-validated, so they are skipped here
+  // absence: validate all inputs normally
+
+  private[expressions] def validateAndParse(methodConfiguration: MethodConfiguration, inputsToParse: Map[String, AttributeString], parser: SlickExpressionParser) = {
```

Comment:
Should this return a ValidatedMethodConfiguration?

Reply:
Yes.  Are you saying you want a return type signature?

Reply:
yes